src/NuGet.Core/NuGet.ProjectModel/CacheFileFormat.cs

```diff
@@ -23,6 +24,30 @@ public static CacheFile Load(string filePath)
             return Load(filePath, NullLogger.Instance);
         }
 
+        public static CacheFile SafeLoad(string filePath, ILogger log)
+        {
+            if (filePath == null)
+            {
+                throw new ArgumentNullException(nameof(filePath));
+            }
+
+            var retries = 3;
+            for (var i = 1; i <= retries; i++)
+            {
+                // Ignore exceptions for the first attempts
+                try
+                {
+                    return Load(filePath, log);
+                }
+                catch (Exception ex) when ((i < retries) && (ex is UnauthorizedAccessException || ex is IOException))
+                {
+                    Thread.Sleep(100);
+                }
+            }
+            // This will never reached, but the compiler can't detect that 
+            return null;
+        }
+
         public static CacheFile Load(string filePath, ILogger log)
         {
             using (var stream = File.OpenRead(filePath))
```

Comment:
Open the stream in the same way this is opened: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.PackageManagement/Projects/PackagesConfigNuGetProject.cs#L253

@jainaashish is fixing this for project.json scenarios currently, he can provide the best settings to use